pkg/util/e2eutil/wait_util.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2eutil
+
+import (
+	"testing"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	retryInterval = time.Second * 5
+	retries       = 6
+)
+
+// WaitForDeployment checks to see if a given deployment has a certain number of available replicas after a specified amount of time
+// If the deployment does not have the required number of replicas after 5 * retries seconds, the function returns an error
+// This can be used in multiple ways, like verifying that a required resource is ready before trying to use it, or to test
+// failure handling, like simulated in SimulatePodFail.
+func WaitForDeployment(t *testing.T, kubeclient kubernetes.Interface, namespace, name string, replicas, retries int) error {
```

Comment:
Let's just pass in the retry interval and total timeout and keep it the same as wait.Poll() expects it. Also that way we don't need to set the global vars for retryInterval and retries on lines 28-29

Reply:
In addition to taking the very good advice to make this stateless, I'd also recommend just keeping it a private function of the example test file until enough copypasta is observed in the wild to justify building up a util package (which tend to get gorpy very fast).